File: 프로그래머스/2/87946. 피로도/피로도.py
'''
최소 필요 피로도, 소모 피로도
''' 
from itertools import permutations
def solution(k, dungeons):
    answer = -1
    L = len(dungeons)

    for lst in list(permutations(dungeons,L)):
        t = k
        cnt = 0
        for mn,use in lst:
            if t>=mn:
                t-=use
                cnt+=1
        answer = max(answer,cnt)
        
    return answer

File: 프로그래머스/2/87946. 피로도/피로도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, vector<vector<int>> dungeons) {
    int answer = 0;
    sort(dungeons.begin(), dungeons.end());

    do{
        int cnt = 0;
        int tmp = k;
        for(int i=0; i<dungeons.size(); i++){
            if(dungeons[i][0] <= tmp){
                cnt += 1;
                tmp -= dungeons[i][1];
            }
        }
        answer = max(answer, cnt);
    }
    while(next_permutation(dungeons.begin(), dungeons.end()));
    
    return answer;
}